src/item_factory.cpp

```diff
@@ -171,11 +181,8 @@ void Item_factory::finalize() {
             }
         }
 
-        for( use_function &use_fun : obj.use_methods ) {
-            iuse_actor *actor = use_fun.get_actor_ptr();
-            if( actor != nullptr ) {
-                actor->finalize( obj.id );
-            }
+        for( auto &e : obj.use_methods ) {
+            e.second.get_actor_ptr()->finalize( obj.id );
```

Comment:
It can return `nullptr` but not for any instance constructed within this module so the check is misleading in that it implies `item` could contain a null actor which is not the case